extensions/load-balancer/gce-extension/src/main/java/org/apache/stratos/gce/extension/GCELoadBalancer.java

```diff
@@ -0,0 +1,391 @@
+        for (String clusterId : clusterToLoadBalancerConfigurationMap.keySet()) {
+            if (!activeClusterIdList.contains(clusterId)) {
+                log.info("Removing the configuration for cluster " + clusterId);
+                deleteConfigurationForCluster(clusterId);
+                clusterToLoadBalancerConfigurationMap.remove(clusterId);
+            }
+        }
+        activeClusterIdList.clear();
+        log.info("Load balancer was configured as given topology");
+        return true;
+    }
+
+    /**
+     * This method is used to delete all forwarding rules, target pools and health checks
+     * in IaaS side according to new topology
+     */
+    private void deleteConfigurationForCluster(String clusterId) throws LoadBalancerExtensionException {
+
+        log.info("Deleting configuration for cluster " + clusterId);
+        GCEClusterConfigurationHolder gceClusterConfigurationHolder = clusterToLoadBalancerConfigurationMap.get(clusterId);
+        //delete forwarding rule
+        gceOperations.deleteForwardingRule(gceClusterConfigurationHolder.getForwardingRuleName());
+        //delete target pool from GCE
+        gceOperations.deleteTargetPool(gceClusterConfigurationHolder.getTargetPoolName());
+        //delete health check from GCE
+        gceOperations.deleteHealthCheck(gceClusterConfigurationHolder.getHealthCheckName());
+        log.info("Deleted configuration for cluster " + clusterId);
+
+    }
+
+    /**
+     * This method is used to create target pools, forwarding rule and a health check related to a cluster
+     * in IaaS side according to new topology
+     */
+    private void createConfigurationForCluster(String clusterId) throws LoadBalancerExtensionException {
+
+        log.info("Creating configuration for cluster");
+
+        GCEClusterConfigurationHolder gceClusterConfigurationHolder = clusterToLoadBalancerConfigurationMap.get(clusterId);
+
+        //create a health check
+        gceOperations.createHealthCheck(gceClusterConfigurationHolder.getHealthCheckName());
+
+        //crate a target pool in GCE
+        gceOperations.createTargetPool(gceClusterConfigurationHolder.getTargetPoolName(),
+                gceClusterConfigurationHolder.getHealthCheckName());
+
+        //add instances to target pool
+        gceOperations.addInstancesToTargetPool(gceClusterConfigurationHolder.getMemberList(),
+                gceClusterConfigurationHolder.getTargetPoolName());
+
+        //create forwarding rules in GCE
+        List<Integer> ipList = gceClusterConfigurationHolder.getIpList();
+        //need to create a port range String
+        String portRange;
+        //if the ip list is empty
+        if (ipList.isEmpty()) {
+            log.warn("Ip list is null");
+            //set all ports to be opened
+            portRange = "1-65535";
+
+        }
+        //else if ip list has only one value
+        else if (ipList.size() == 1) {
```

Comment:
Hi Imesh,
In GCE forwarding rules, we should specify ports as ranges. If we need  to forward only one port like 80, we should give it as "80-80". That is the default format they are using.
so we have 3 cases.
1. no any port specified - use the default value
2. only one port to be forwarded - repeat the same port twice
3. set of ports to be forwarded - Sort the port numbers in ascending order and use the lowest one and highest one. So all ports in between lowest and highest will be forwarded